From: =?utf-8?b?Ik1hcmNvIFRyZXZpc2FuIChUcmV2acOxbyki?= <mail@3v1n0.net>
Date: Fri, 20 Mar 2026 18:32:25 +0100
Subject: applications-panel: Use portal app ID to store portal settings

While the app-ID in flatpak world matches the portal permission ID, this
not true for snaps, where we should instead use the portal ID.

So use this value instead

Origin: https://gitlab.gnome.org/GNOME/gnome-control-center/-/merge_requests/3353
---
 panels/applications/cc-applications-panel.c | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/panels/applications/cc-applications-panel.c b/panels/applications/cc-applications-panel.c
index 3f597fa..ae31f34 100644
--- a/panels/applications/cc-applications-panel.c
+++ b/panels/applications/cc-applications-panel.c
@@ -454,7 +454,7 @@ set_notification_allowed (CcApplicationsPanel *self, gboolean allowed)
 static void
 notification_cb (CcApplicationsPanel *self)
 {
-    if (self->current_app_id)
+    if (self->notification_settings || self->current_portal_app_id)
         set_notification_allowed (self, adw_switch_row_get_active (self->notifications_row));
 }
 
@@ -506,7 +506,7 @@ set_background_allowed (CcApplicationsPanel *self, gboolean allowed)
 static void
 background_cb (CcApplicationsPanel *self)
 {
-    if (self->current_app_id)
+    if (self->current_portal_app_id)
         set_background_allowed (self, adw_switch_row_get_active (self->background_row));
 }
 
@@ -527,13 +527,13 @@ set_wallpaper_allowed (CcApplicationsPanel *self, gboolean allowed)
     const gchar *perms[2] = { NULL, NULL };
 
     perms[0] = allowed ? "yes" : "no";
-    set_portal_permissions (self, "wallpaper", "wallpaper", self->current_app_id, perms);
+    set_portal_permissions (self, "wallpaper", "wallpaper", self->current_portal_app_id, perms);
 }
 
 static void
 wallpaper_cb (CcApplicationsPanel *self)
 {
-    if (self->current_app_id)
+    if (self->current_portal_app_id)
         set_wallpaper_allowed (self, adw_switch_row_get_active (self->wallpaper_row));
 }
 
@@ -554,13 +554,13 @@ set_screenshot_allowed (CcApplicationsPanel *self, gboolean allowed)
     const gchar *perms[2] = { NULL, NULL };
 
     perms[0] = allowed ? "yes" : "no";
-    set_portal_permissions (self, "screenshot", "screenshot", self->current_app_id, perms);
+    set_portal_permissions (self, "screenshot", "screenshot", self->current_portal_app_id, perms);
 }
 
 static void
 screenshot_cb (CcApplicationsPanel *self)
 {
-    if (self->current_app_id)
+    if (self->current_portal_app_id)
         set_screenshot_allowed (self, adw_switch_row_get_active (self->screenshots_row));
 }
 
